test/common/event/scaled_range_timer_manager_impl_test.cc

```diff
@@ -316,8 +316,54 @@ TEST_F(ScaledRangeTimerManagerTest, SingleTimerSameMinMax) {
   EXPECT_FALSE(timer->enabled());
 }
 
+TEST_F(ScaledRangeTimerManagerTest, MaxOnlyScaleFactorTimer) {
+  ScaledRangeTimerManagerImpl manager(dispatcher_);
+
+  // With the default scale factor of 1, the timer should fire at its maximum timeout.
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(ScaleFactor(0.5), callback.AsStdFunction());
+  EXPECT_CALL(callback, Call());
+
+  timer->enableTimer(std::chrono::seconds(10));
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  EXPECT_FALSE(timer->enabled());
+
+  // Setting the scale factor to 0 should result in the timer honoring its minimum timeout, which is
+  // 0.5 * max.
+  manager.setScaleFactor(0);
+  EXPECT_CALL(callback, Call());
+
+  timer->enableTimer(std::chrono::seconds(10));
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, MaxOnlyAbsoluteValueTimer) {
+  ScaledRangeTimerManagerImpl manager(dispatcher_);
+
+  // With the default scale factor of 1, the timer should fire at its maximum timeout.
+  MockFunction<TimerCb> callback;
+  auto timer =
+      manager.createTimer(AbsoluteValue(std::chrono::seconds(3)), callback.AsStdFunction());
```

Comment:
Remind me, what happens when the max is smaller than an absolute min?  Which of the two values wins?  It seems to me that the max should win since it's the one that is more explicitly configured.

Reply:
My read of RangeTimerImpl::enableTimer is that the min is honored, not the max.